tools/python/gen_opkernel_doc.py

```diff
@@ -132,7 +131,7 @@ def main(args):  # type: (Type[Args]) -> None
                                 fout.write('|')
 
                             tclist = []
-                            for tc in tcset:
+                            for tc in sorted(tcset):
```

Comment:
Same explanation as above. Sort so that the type order is not ambiguous. Without this we would see `tensor(double), tensor(float)` or `tensor(float), tensor(double)`

Reply:
Does the gen_doc.py In ONNX repo have the same logic?

Reply:
Not sure. I can check. But I would presume it has (or equivalently stores the non-string version (MLDataType ?) in a fashion that doesn’t make the order ambiguous.

Reply:
In ONNX, the type constraints are stored in a sequence (order is pre-defined an unambiguous) - https://github.com/onnx/onnx/blob/f1fcc92d203f54e87a8a7ee06c51640f3f659974/onnx/defs/gen_doc.py#L193. In this implementation, type constraints were being stored in a Python set -  https://github.com/microsoft/onnxruntime/blob/ec4f6c099b3d3e2ea91106809b8979753af3a3e9/tools/python/gen_opkernel_doc.py#L111 (which can provide data back in any order). Hence, sorting is needed to prevent this. Moreover, there are plenty of such sorting done to prevent such occurrences in the same script. It was just getting missed in a couple of places.